healsparse/geom.py

```diff
@@ -0,0 +1,668 @@
+
+    smap = HealSparseMap.makeEmpty(
+        nsideCoverage=32,
+        nsideSparse=nside,
+        dtype=dtype,
+        sentinel=0,
+    )
+    realize_geom(circles, smap)
+
+    test, = np.where(smap._sparseMap > 0)
+    print(test.size)
+
+    w, = np.where(smap._sparseMap != smap._sentinel)
+    print(w.size)
+
+    data = smap.getValuePixel(smap.validPixels)
+    print('_sparseMap:', smap._sparseMap)
+    print('data:', data)
+    w, = np.where(data != smap._sentinel)
+    print(w.size)
+
+    plt = None
+    if show:
+        import biggles
+        import pcolors
+
+        nrand = 100000
+        rra = rng.uniform(low=ra_range[0], high=ra_range[1], size=nrand)
+        rdec = rng.uniform(low=dec_range[0], high=dec_range[1], size=nrand)
+
+        vals = smap.getValueRaDec(rra, rdec)
+        uvals = np.unique(vals)
+        colors = list(reversed(pcolors.rainbow(uvals.size*2)))
+        print('unique vals:', uvals)
+
+        xrng = ra_range
+        yrng = dec_range
+        aspect = (yrng[1]-yrng[0])/(xrng[1]-xrng[0])
+        plt = biggles.FramedPlot(
+            xrange=xrng,
+            yrnage=yrng,
+            aspect_ratio=aspect,
+            xlabel='RA',
+            ylabel='DEC',
+        )
+
+        for i, val in enumerate(uvals):
+            if val == 0:
+                continue
+            w, = np.where(vals == val)
+            color = colors[i]
+            pts = biggles.Points(rra[w], rdec[w], type='dot', color=color)
+            plt.add(pts)
+
+        plt.show()
+
+    if show_mat:
+        from .visu_func import hsp_view_map
+
+        extent = [
+            ra_range[0]*0.9,
+            ra_range[1]*1.1,
+            dec_range[0]*1.1,
+            dec_range[1]*1.1,
+        ]
+        hsp_view_map(
+            smap,
+            savename='/tmp/test.png',
+            show_coverage=False,
+            extent=extent,
+        )
+
+    return plt
+
+def test_box(show=False):
+    ra = [200.0, 200.2, 200.2, 200.0]
+    dec = [0.0, 0.0, 0.1, 0.1]
+    nside = 2**15
+    poly = Polygon(
+        ra=ra,
+        dec=dec,
+        value=2**4,
+    )
+
+    smap = poly.get_map(nside=nside, dtype=np.int16)
+    print(smap)
+
+    if show:
+        import biggles
+        pixels = poly.get_pixels(nside=nside)
+        pra, pdec = hp.pix2ang(nside, pixels, nest=True, lonlat=True)
+        plt = biggles.plot(
+            pra,
+            pdec,
+            type='filled circle',
+            xlabel='RA',
+            ylabel='DEC',
+            aspect_ratio=0.5,
+            visible=False,
+        )
+        plt.add(
+            biggles.Box((ra[0], dec[0]), (ra[2], dec[2]), color='red'),
+        )
+        plt.show()
+
+def _make_poly():
+    # counter clockwise
+    ra = [200.0, 200.2, 200.3, 200.2, 200.1]
+    dec = [0.0,     0.1,   0.2,   0.25, 0.13]
+    poly = Polygon(
+        ra=ra,
+        dec=dec,
+        value=64,
+    )
+    return poly, ra, dec
+
+def test_polygon(show=False):
+
```

Comment:
Also, is this in tests now?

Reply:
yes, but not the visuallization part

maybe leave for now since these are the only examples of fast lightweight visualization